src/dialogs/more-info/components/siren/ha-more-info-siren-advanced-controls.ts

```diff
@@ -0,0 +1,225 @@
+
+  public showDialog({ stateObj }: { stateObj: HassEntity }) {
+    this._stateObj = stateObj;
+  }
+
+  public closeDialog(): void {
+    this._dialog?.close();
+  }
+
+  private _dialogClosed(): void {
+    this._stateObj = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  render() {
+    if (!this._stateObj) {
+      return nothing;
+    }
+    const supportsTones =
+      supportsFeature(this._stateObj, SirenEntityFeature.TONES) &&
+      this._stateObj.attributes.available_tones;
+    const supportsVolume = supportsFeature(
+      this._stateObj,
+      SirenEntityFeature.VOLUME_SET
+    );
+    const supportsDuration = supportsFeature(
+      this._stateObj,
+      SirenEntityFeature.DURATION
+    );
+    return html`
+      <ha-md-dialog
+        open
+        @closed=${this._dialogClosed}
+        aria-labelledby="dialog-light-color-favorite-title"
+        .getOpenAnimation=${getMobileOpenFromBottomAnimation}
+        .getCloseAnimation=${getMobileCloseToBottomAnimation}
+      >
+        <ha-dialog-header slot="headline">
+          <ha-icon-button
+            slot="navigationIcon"
+            @click=${this.closeDialog}
+            .label=${this.hass.localize("ui.common.close")}
+            .path=${mdiClose}
+          ></ha-icon-button>
+          <span slot="title" id="dialog-light-color-favorite-title"
+            >${this.hass.localize(
+              "ui.components.siren.advanced_controls"
+            )}</span
+          >
+        </ha-dialog-header>
+        <div slot="content">
+          <div class="options">
+            ${supportsTones
+              ? html`
+                  <ha-select
+                    .label=${this.hass.localize("ui.components.siren.tone")}
+                    @closed=${stopPropagation}
+                    @change=${this._handleToneChange}
+                    .value=${this._tone}
+                  >
+                    <ha-list-item value="">
+                      ${this.hass.localize("ui.common.default")}
+                    </ha-list-item>
+                    ${Object.entries(
+                      this._stateObj.attributes.available_tones
+                    ).map(
+                      ([toneId, toneName]) => html`
+                        <ha-list-item .value=${toneId}
+                          >${toneName}</ha-list-item
+                        >
+                      `
+                    )}
+                  </ha-select>
+                `
+              : nothing}
+            ${supportsVolume
+              ? html`
+                  <ha-textfield
+                    type="number"
+                    .label=${this.hass.localize("ui.components.siren.volume")}
+                    .value=${this._volume}
+                    @change=${this._handleVolumeChange}
+                    .min=${0}
+                    .max=${1}
+                    .step=${0.1}
+                  ></ha-textfield>
```

Comment:
I would do here 0-100 and convert it on play.

Or what about adding a range slider default to 50%?